src/rmarkdown/knit.ts

```diff
@@ -0,0 +1,250 @@
+import * as util from '../util';
+import * as vscode from 'vscode';
+import * as fs from 'fs-extra';
+import path = require('path');
+import yaml = require('js-yaml');
+
+import { RMarkdownManager, KnitWorkingDirectory } from './manager';
+import { runTextInTerm } from '../rTerminal';
+
+export let knitDir: string = util.config().get<string>('rmarkdown.knit.defaults.knitWorkingDirectory') ?? undefined;
```

Comment:
Preferrably this should be typed as `KnitWorkingDirectory` (to make clear that it does not contain e.g. a literal path)